### ISSUE TITLE
An improvement on QuickSort

### ISSUE THREAD

**Issue by 1170300803** [MauriceGit/Advanced_Algorithms#3]
Quick sort algorithm can be optimized in many ways. For example, instead of selecting the last element, the benchmark element can randomly choose one, which avoids the case of an ordered array with the worst time complexity of log (n2).

**Comment by MauriceGit** [comment]
True. Feel free to implement that one change and create a pull-request.
Maybe I find time for this change myself otherwise.

**Comment by MauriceGit** [comment]
Just to note: Quicksort always has a worst-case time complexity of O(n²), independent of the implementation detail of pivot element choice. Average runtime changes to log n for special cases.
